File: benchmarks/fixtures/webapp_swift/Sources/webapp/util/Logger.swift
import Foundation

/// Severity level for log records.
enum LogLevel: Int {
    case debug, info, warn, error
}

/// Simple stdout logger shared across services.
final class Logger {
    let name: String

    init(name: String) {
        self.name = name
    }

    func info(_ message: String) {
        emit(.info, message)
    }

    func warn(_ message: String) {
        emit(.warn, message)
    }

    func error(_ message: String) {
        emit(.error, message)
    }

    private func emit(_ level: LogLevel, _ message: String) {
        print("[\(name)] \(message)")
    }
}

/// Return a logger tagged with the caller's component name. High-fanout helper:
/// nearly every service obtains its logger through this function.
func getLogger(_ component: String) -> Logger {
    return Logger(name: component)
}

File: benchmarks/fixtures/webapp_kt/src/main/kotlin/webapp/util/Logger.kt
package webapp.util

/** Severity level for log records. */
enum class LogLevel {
    DEBUG,
    INFO,
    WARN,
    ERROR
}

/** Simple stdout logger shared across services. */
class Logger(val name: String) {

    fun info(message: String) {
        emit(LogLevel.INFO, message)
    }

    fun warn(message: String) {
        emit(LogLevel.WARN, message)
    }

    fun error(message: String) {
        emit(LogLevel.ERROR, message)
    }

    private fun emit(level: LogLevel, message: String) {
        println("[$name] $message")
    }
}

/**
 * Return a logger tagged with the caller's component name. High-fanout helper:
 * nearly every service obtains its logger through this function.
 */
fun getLogger(component: String): Logger {
    return Logger(component)
}
